### PR TITLE
Converted to Spring boot 2 to enable WebClient tracing instrumentation

### PR DESCRIPTION
**Pull request by mykyta-protsenko**

### CODE REVIEW

File: gateway/build.gradle [hunk 1]
```diff
@@ -22,17 +31,32 @@ def grpcVersion = '1.14.0'
...
+    imports {
+        mavenBom 'org.springframework.cloud:spring-cloud-sleuth:2.0.2.BUILD-SNAPSHOT'
+    }
 }
 
 dependencies {
     compileOnly "org.immutables:value:2.7.1:annotations"
     annotationProcessor "org.immutables:value:2.7.1"
 
+    //compile('org.springframework.boot:spring-boot-web-reactive')
```
Comment (alxbnet) [inline gateway/build.gradle]: Remove commented out?

File: gateway/build.gradle [hunk 2]
```diff
@@ -61,18 +88,23 @@ apply plugin: 'application'
...
-
-    from {
-        configurations.compile.collect { it.isDirectory() ? it : zipTree(it) }
-    }
+bootJar {
+    mainClassName = "${mainClass}"
+    version = ''
 }
 
+//jar {
```
Comment (alxbnet) [inline gateway/build.gradle]: Remove?

File: gateway/src/main/java/xyz/breakit/gateway/Gateway.java [hunk 3]
```diff
@@ -1,49 +1,114 @@
...
+
+        String zipkinHost = System.getenv().getOrDefault("ZIPKIN_SERVICE_HOST", "zipkin");
+        int zipkinPort = Integer.valueOf(System.getenv().getOrDefault("ZIPKIN_SERVICE_PORT", "9411"));
+
+        URLConnectionSender sender = URLConnectionSender.newBuilder()
+                .endpoint(String.format("http://%s:%s/api/v2/spans", zipkinHost, zipkinPort))
+                .build();
+
+        return GrpcTracing.create(Tracing.newBuilder()
+                .sampler(ALWAYS_SAMPLE)
```
Comment (alxbnet) [inline gateway/src/main/java/xyz/breakit/gateway/Gateway.java]: Inject Sampler bean instead?

File: gateway/src/main/java/xyz/breakit/gateway/GatewayClient.java [hunk 4]
```diff
@@ -12,7 +12,12 @@
 
     public static void main(String[] args) {
         String host = "35.230.13.179";
-        Channel channel = ManagedChannelBuilder.forAddress(host, 8080).usePlaintext().build();
+
```
Comment (alxbnet) [inline gateway/src/main/java/xyz/breakit/gateway/GatewayClient.java]: Please remove an extra line

File: gateway/src/main/java/xyz/breakit/gateway/clients/leaderboard/LeaderboardClient.java [hunk 5]
```diff
@@ -1,20 +1,33 @@
...
 import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.ScheduledThreadPoolExecutor;
 import java.util.concurrent.TimeUnit;
 
+@Service
 public class LeaderboardClient {
 
+    public static final Logger LOG = LoggerFactory.getLogger(LeaderboardClient.class);
+
     // one per core
```
Comment (alxbnet) [inline gateway/src/main/java/xyz/breakit/gateway/clients/leaderboard/LeaderboardClient.java]: Please use Runtime.getRuntime().availableProcessors() instead.

File: gateway/build.gradle (hunk 2 above)
Comment (mykyta-protsenko) [reply gateway/build.gradle]: Done

**Pull request merged**